cloudinit/config/cc_mounts.py

```diff
@@ -57,7 +55,7 @@ def is_meta_device_name(name):
 
 def is_network_device(name):
     # return true if this is a network device
-    if NETWORK_NAME_RE.match(name):
+    if re.compile(NETWORK_NAME_FILTER).match(name):
```

Comment:
> This can just be an `re.match()` because compiling a regex is to pay the time cost once so each subsequent match can be faster.

+1

> If the time cost of not being compiled is significant, it may be better to move the compilation into a memoized initialization function. Same comment for the rest of the regexes in this PR.

Every call to `re.match()` [fetches previously-compiled expressions](https://github.com/python/cpython/blob/78cb377c622a98b1bf58df40c828e886575a6927/Lib/re/__init__.py#L330) under the covers, so I think this memoization is probably handled for us.